core/src/test/java/org/elasticsearch/common/settings/loader/JsonSettingsLoaderTests.java

```diff
@@ -48,6 +49,10 @@ public void testSimpleJsonSettings() throws Exception {
     }
 
     public void testDuplicateKeysThrowsException() {
+        if (JsonXContent.isStrictDuplicateDetectionEnabled()) {
```

Comment:
Oh, I didn't know about `assumeTrue` / `assumeFalse` and have only noticed the approach that I've used in the code base. Btw, I think this should be `assumeFalse`. But this makes sense and I'll change the tests as you've suggested. Thanks for the pointer.